Classes/Controller/UserController.php

```diff
@@ -56,6 +57,10 @@ public function saveSettingsAction(FrontendUser $frontendUser) {
             }
         }
 
+        if ($frontendUser->getUserRole() != Security::ROLE_LIBRARIAN) {
```

Comment:
I don't get this check. API-Tokens are only given out for users with the `ROLE_LIBRARIAN` and yet we check here if the user has a different role and reset the token to an empty string. What is the purpose of this check?